diag_manager/fms_diag_yaml_object.F90

```diff
@@ -137,10 +114,9 @@ module fms_diag_yaml_object_mod
      character (len=:), allocatable :: var_outname !< Name of the variable as written to the file
      character (len=:), allocatable :: var_longname !< Overwrites the long name of the variable
      character (len=:), allocatable :: var_units !< Overwrites the units
-     character (len=:), dimension (:), allocatable :: var_attributes !< Attributes to overwrite or
+     character (len=MAX_STR_LEN), dimension (:, :), allocatable :: var_attributes !< Attributes to overwrite or
```

Comment:
I think you should add a comment about why `len=MAX_STR_LEN` and not `:` for someone else who will have to deal with this in the future.